teuthology/suite/run.py

```diff
@@ -516,13 +520,47 @@ def schedule_suite(self):
         ).name
         self.base_yaml_paths.insert(0, base_yaml_path)
 
+        # compute job limit in respect of --sleep-before-teardown
+        job_limit = self.args.limit or 0
+        sleep_before_teardown = int(self.args.sleep_before_teardown or 0)
+        if sleep_before_teardown:
+            if job_limit == 0:
+                log.warning('The --sleep-before-teardown is provided, '
+                            'only 1 job will be scheduled. '
+                            'Use --limit to run more jobs')
+                # give user a moment to read this warning
+                time.sleep(5)
+                job_limit = 1
+            elif self.args.non_interactive:
+                log.warning(
+                    'The --sleep-before-teardown is used and there will be maximum '
+                    '{} jobs running which will fall asleep for {}'
+                    .format(job_limit, format_timespan(sleep_before_teardown)))
+            elif job_limit > 4:
+                are_you_insane=(
+                    'There are {total} configs and {maximum} job limit is used. '
+                    'Do you really want to lock all machines needed for '
+                    'this run for {that_long}? Are you insane? (y/N):'
```

Comment:
I suggest to drop the additional "Are you insane?" question. The first question is sufficient, no?

Reply:
:-)